engine/source/public/core/asset.hpp

```diff
@@ -0,0 +1,70 @@
+﻿#pragma once
+#include <string>
+
+namespace lumina
+{
+    enum class AssetType
+    {
+        Texture, // Including Samplers or not can be discussed
+        Shader,
+        Model, // Collections of meshes
+        Mesh,
+        Material,
+        None
+    };
+
+    class Asset
+    {
+        friend class AssetManager;
+
+    private:
+        static size_t nextID;
+
+    protected:
+        size_t id;
+        AssetType type;
+        std::string filePath;
+
+        Asset(AssetType type) : id(0), type(type) {}
+
+        virtual ~Asset();
+
+        virtual void Reload() {}
+
+        static size_t GenerateID(const std::string& path)
+        {
+            return std::hash<std::string> {}(path);
```

Comment:
is this line correct? `{}(path);`

Reply:
Yes. It is a brace initialization, C++ 11 introduced it, {} prevents narrowing conversion. It is just more consistent way of doing this.

Reply:
I had never seen this before. I don't know about narrowing conversion, but from what I found, it is the exact same as:
I don't this has impact on narrowing conversion, but that it is simply a shorter way of writing the has by creating a temporary object from hash and calling it with the parameter.
```c++
static size_t GenerateID(const std::string& path)
{
    std::hash<std::string> string_hash;
    return string_hash(path);
}
```

Reply:
I feel like my implementation is cleaner, although the {} has a space because of the formatting. I don't know what you think?

Reply:
your implementation is definitly cleaner, i would keep it. But this was more of an explanation to myself on what is actually happening, so i just wrote it out to show what i found in case you weren't 100% yourself :D